import{_ as e}from"./plugin-vue_export-helper-DlAUqK2U.js";import{c as t,o as r,d as o}from"./app-D5166FK3.js";const l={},i=o('<h2 id="异常" tabindex="-1"><a class="header-anchor" href="#异常"><span>异常</span></a></h2><p>异常是继承Throwble类的Exception和Error</p><ul><li>Throwble <ul><li>Error：OutOfMemoryError、NoClassDefFoundError</li><li>Exception <ul><li>Checked Exception：程序不处理（catch-throw）不能通过编译；FileNotFoundException</li><li>Unchecked Exception：RuntimeException及其子类（NullPointerException、ArrayIndexOutOfBoundsException）</li></ul></li></ul></li></ul><h3 id="常用方法" tabindex="-1"><a class="header-anchor" href="#常用方法"><span>常用方法</span></a></h3><ul><li>getMessage()</li><li>toString()</li><li>printStackTrace()</li><li>getLocalizedMessage()</li></ul><h3 id="try-catch-finally的执行情况" tabindex="-1"><a class="header-anchor" href="#try-catch-finally的执行情况"><span>try-catch-finally的执行情况</span></a></h3><ul><li>try-catch</li><li>try-finally</li></ul><p><code>finally</code>：</p><ul><li>无论是否捕获或处理异常，<code>finally</code> 块里的语句都会被执行，除非finally 之前虚拟机被终止运行，比如在catch中程序中断（System.exit(1);）程序所在的线程死亡；关闭 CPU</li><li>在 <code>try</code> 块或 <code>catch</code> 块中有 <code>return</code> 语句时，先执行<code>finally</code> 语句块再执行return</li><li>不要在finally中用return：如果finally中有return，try中的return语句会被忽略，返回finally中的值</li></ul><p>面对异常有两种处理：</p><ol><li>抛出不catch处理：throw，方法上是throws</li><li>catch处理；抛出最终也会处理，如果自己没有写代码处理，就是JVM最后catch并处理</li></ol><h2 id="泛型generics" tabindex="-1"><a class="header-anchor" href="#泛型generics"><span>泛型<strong>Generics</strong></span></a></h2><p>编译器对泛型参数进行检测、自动转换为目标对象</p><p>静态方法无法使用类中的泛型，因为静态方法的加载优先于类的实例化</p>',14),c=[i];function n(a,p){return r(),t("div",null,c)}const d=e(l,[["render",n],["__file","3. 异常泛型.html.vue"]]),u=JSON.parse(`{"path":"/backend/java/3.%20%E5%BC%82%E5%B8%B8%E6%B3%9B%E5%9E%8B.html","title":"异常泛型","lang":"zh-CN","frontmatter":{"title":"异常泛型","order":3,"copyright":"<a href=\\"https://creativecommons.org/licenses/by-nc/4.0/\\">CC BY-NC 4.0协议</a>","description":"异常 异常是继承Throwble类的Exception和Error Throwble Error：OutOfMemoryError、NoClassDefFoundError Exception Checked Exception：程序不处理（catch-throw）不能通过编译；FileNotFoundException Unchecked Excep...","head":[["meta",{"property":"og:url","content":"https://vuepress-theme-hope-docs-demo.netlify.app/KKBlog/backend/java/3.%20%E5%BC%82%E5%B8%B8%E6%B3%9B%E5%9E%8B.html"}],["meta",{"property":"og:site_name","content":"KK's Blog"}],["meta",{"property":"og:title","content":"异常泛型"}],["meta",{"property":"og:description","content":"异常 异常是继承Throwble类的Exception和Error Throwble Error：OutOfMemoryError、NoClassDefFoundError Exception Checked Exception：程序不处理（catch-throw）不能通过编译；FileNotFoundException Unchecked Excep..."}],["meta",{"property":"og:type","content":"article"}],["meta",{"property":"og:locale","content":"zh-CN"}],["meta",{"property":"og:updated_time","content":"2024-05-14T07:00:16.000Z"}],["meta",{"property":"article:author","content":"KK"}],["meta",{"property":"article:modified_time","content":"2024-05-14T07:00:16.000Z"}],["script",{"type":"application/ld+json"},"{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"Article\\",\\"headline\\":\\"异常泛型\\",\\"image\\":[\\"\\"],\\"dateModified\\":\\"2024-05-14T07:00:16.000Z\\",\\"author\\":[{\\"@type\\":\\"Person\\",\\"name\\":\\"KK\\",\\"url\\":\\"https://github.com/Nikki-Gu\\"}]}"]]},"headers":[{"level":2,"title":"异常","slug":"异常","link":"#异常","children":[{"level":3,"title":"常用方法","slug":"常用方法","link":"#常用方法","children":[]},{"level":3,"title":"try-catch-finally的执行情况","slug":"try-catch-finally的执行情况","link":"#try-catch-finally的执行情况","children":[]}]},{"level":2,"title":"泛型Generics","slug":"泛型generics","link":"#泛型generics","children":[]}],"git":{"createdTime":1715670016000,"updatedTime":1715670016000,"contributors":[{"name":"Nikki-Gu","email":"394632208@qq.com","commits":1}]},"readingTime":{"minutes":1.01,"words":302},"filePathRelative":"backend/java/3. 异常泛型.md","localizedDate":"2024年5月14日","autoDesc":true}`);export{d as comp,u as data};
